core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -368,35 +392,50 @@ public void onFailure(Throwable e) {
 
         @Override
         protected void doRun() {
+            // shardID added to the request, as a side effect of calling resolveRequest
             if (checkBlocks() == false) {
```

Comment:
can we change the method name to resolveRequestAndCheckBlocks? I had a worry that the request is not resolved by the time we check the index routing table based on request.shardId.